cmd/jujud/agent/bootstrap.go

```diff
@@ -190,43 +256,54 @@ func (c *BootstrapCommand) Run(_ *cmd.Context) error {
 		return nil
 	}
 
-	instances, err := env.Instances(callCtx, []instance.Id{args.BootstrapMachineInstanceId})
-	if err != nil {
-		return errors.Annotate(err, "getting bootstrap instance")
-	}
-	addrs, err := instances[0].Addresses(callCtx)
-	if err != nil {
-		return errors.Annotate(err, "bootstrap instance addresses")
+	info, ok := agentConfig.StateServingInfo()
+	if !ok {
+		return fmt.Errorf("bootstrap machine config has no state serving info")
 	}
 
-	// When machine addresses are reported from state, they have
-	// duplicates removed.  We should do the same here so that
-	// there is not unnecessary churn in the mongo replicaset.
-	// TODO (cherylj) Add explicit unit tests for this - tracked
-	// by bug #1544158.
-	addrs = network.MergedAddresses([]network.Address{}, addrs)
+	addrs := []network.Address{network.NewAddress("127.0.0.1")}
+	if !isCAAS {
```

Comment:
Can we break this block of code out to a separate function to save all the indenting etc

Reply:
sure, moved to a separate func.